dbms/src/Server/Server.cpp

```diff
@@ -1135,6 +1134,11 @@ int Server::main(const std::vector<std::string> & /*args*/)
         global_context->getPathCapacity(),
         global_context->getFileProvider());
 
+    /// Initialize the background thread pool.
+    /// set the background pool & blockable background pool size to the a quarter of of number of logical CPU cores of machine.
+    auto & bg_pool = global_context->initializeBackgroundPool(server_info.cpu_info.logical_cores / 4);
```

Comment:
Use `Settings.background_pool_size` instead of hardcode here.

Set `Settings.background_pool_size`'s default value to `server_info.cpu_info.logical_cores / 4`.

Reply:
done